### ISSUE TITLE
Negative Ridge Regression Outputs

### ISSUE THREAD

**Issue by alesroger** [mahmoodlab/HEST#102]
Hi dear authors. Firstly, great work! I want to use the method you proposed to predict gene expression (PCA+Ridge Regression). I notice (on my data at the moment) that also negative gene expression values are predicted (even though we want to predict log1p, i.e. all targets are >= 0). PCC can still be relatively high with negative predicted values as well. How did you handle this and what are your thoughts about this? Am I missing something or some details in the code where you prevent this from happening? Thank you so much for your help!

**Comment by pauldoucet** [comment]
Dear @alesroger, thank you for your comment,

### Regarding PCC as a metric:
Indeed, PCC might still be high even if predicted expression values are negative, more generally PCC is invariant [under separate changes in location and scale in the two variables](https://en.wikipedia.org/wiki/Pearson_correlation_coefficient) -> meaning that predicted values might have a different slope than real expression value and still yield the same PCC.

In our case the goal is to predict whenever foundation models can correctly predict up-and-down variations in expression, i.e. “When the real expression goes up, does my model's prediction also go up (and by how much)?”. Indeed, in that case, MSE could be added in addition to PCC to evaluate the absolute difference between predicted and real expression.

### Regarding linear regression

However, we chose not to penalize negative expression predictions as linear regression is a naive model used solely for the sake of benchmarking. If the goal were to maximize performance in a real-world application, a non-linear model will have no problem in correctly predicting the sign of the expression. We use linear regression during benchmarking for two main reasons:

1) Replacing the linear projection head with an MLP would require tuning multiple hyperparameters for each model which makes fair comparison between models harder, instead we simply assume that the performance with a linear regression correlates positively with that of an MLP.
2) FMs are off-the-shelf learners, often used together with a light projection head -> therefore we’re also interested in measuring the quality of the embedding space: i.e. “how easily can light models learn from this FM’s embeddings”. 

Such negative expression penalization can be implemented when training the actual production projection head, it can then be baked into the loss function or directly in the project head architecture.

**Comment by alesroger** [comment]
thank you for your detailed answer!
